cmd/template/app/runner.go

```diff
@@ -18,17 +18,17 @@ import (
 	templateapp "github.com/giantswarm/kubectl-gs/pkg/template/app"
 )
 
-type runner struct {
-	flag   *flag
-	logger micrologger.Logger
-	stdout io.Writer
-	stderr io.Writer
+type Runner struct {
```

Comment:
Rather than exporting the runner I'd prefer more of the logic is moved to `pkg/template/app`.
Can you take a look and see how hard that would be?

Reason being that multiple teams need to work on `kubectl gs` so it's good to keep the code as consistent as possible. 

Also me may need to support more types of apps in future. e.g. an app of apps for the security stack.

Reply:
@rossf7  Agree that exporting apps introduced some inconsistency, it may hurt aesthetics 😄 I tried to re-organize the code a bit, to move some parts to the `package/template/app` and thus avoid that much code repetition, though I failed to fully accomplish the goal. At the moment the `pkg/template/app` provides some structs and functions, we relay on the returned values to modify instances of these structs in the `cmd/template/app`, that's why a good dose of code lives in the `cmd/template/app` and the `cmd/template/cluster/provider/capo`. I believe if we want to move more of it into the `package/template/app` we will need to make classes out of these structs, and make the functions to be their methods, but I am not sure if that's not too big of a change.

Reply:
Thx! The structure looks a lot better ❤️ 

The main thing was I didn't want the runner to be exported. I think this is fine for now we can always revisit when we add support for more providers.

Reply:
Perfect, good to hear that, I made the suggested changes and fixed import so that CI's happy.